examples/ublox_receiver_plugin/ublox_receiver_view.cpp

```diff
@@ -0,0 +1,86 @@
+#include "ublox_receiver_view.h"
+
+#include "receiver_enums.h"
+#include "ui_ublox_receiver_view.h"
+
+UbloxReceiverView::UbloxReceiverView(QWidget* parent) : QWidget(parent), ui(new Ui::UbloxReceiverView)
+{
+  ui->setupUi(this);
+
+  selectedStartType = ReceiverStartType::NONE;
+  setReceiverStatus(ReceiverStatus::NOT_CONNECTED);
+
+  connect(ui->startButton, &QPushButton::clicked, [this] {
+    ui->startButton->setEnabled(false);
+    emit startClicked(this->selectedStartType);
+  });
+
+  connect(ui->startTypeSelect, qOverload<int>(&QComboBox::currentIndexChanged), [this](int index) {
+    this->selectedStartType = (ReceiverStartType)index;
+    if (index == 0)
+    {
+      ui->startButton->setEnabled(false);
+    }
+    else if (receiverStatus == ReceiverStatus::ACTIVE || receiverStatus == ReceiverStatus::INACTIVE)
+    {
+      ui->startButton->setEnabled(true);
+    }
```

Comment:
What happens when you are in the else case?

Reply:
The else case is when the receiverStatus is in a state where we cannot start the receiver (like NOT_CONNECTED), so we would not want to enable the startButton